### ISSUE TITLE
Clicking stop will erase the progress

### ISSUE THREAD

**Issue by jojorne** [LostRuins/koboldcpp#1074]
**Describe the Issue**
Click stop generating before it finishes. It will erase everything that has been streamed so far. This bug has been fixed, but it is back.

**Additional Information:**
Windows 11, Opera, Edge, Firefox, koboldcpp-1.72

**Comment by LostRuins** [comment]
Kobold Lite frontend? I don't suppose trim sentences is enabled?

**Comment by jojorne** [comment]
> Kobold Lite frontend? I don't suppose trim sentences is enabled?

![image](https://github.com/user-attachments/assets/6ee96dcb-bb85-4492-b48b-a5d504b866be)

**Comment by LostRuins** [comment]
I cannot repro it. Can you link me back to the previous issue where this was resolved so I can review what was done?

**Comment by jojorne** [comment]
I think 98dff80b9c5a4e0424230245dd99b98ed3b5b037 fixed it. I'll keep testing though.

**Comment by LostRuins** [comment]
Can you see if the latest release 1.73 still has this issue?

**Comment by VL4DST3R** [comment]
I've also encountered this bug on and off from time to time, although in my case it seems to lose only incomplete sentences (in all 4 modes), usually up to the previous period or new line. I also have trim sentences disabled and 'Token Streaming' set to `Poll`.

**Comment by jojorne** [comment]
Every time it outputs a `,` with it as a stop seq, text generated disappear.

Edit: I have XTC and DRY on.

**Comment by LostRuins** [comment]
> Every time it outputs a , with it as a stop seq, text generated disappear.

Do you mean all the text after and including the `,` disappears? Or everything in the output disappears?

**Comment by jojorne** [comment]
My quick test prompt:
```

{{[INPUT]}}
I need you to continue the following list:

**Fruits**

banana, apple,
{{[OUTPUT]}}
Here is the continued list:

**Fruits**
```
I think I found the reason:
Set stop sequence to ',' or even 'apple' and then usage mode to "Instruct Mode".
It will trigger the `end_trim_to_sentence` cleaning the whole paragraph.
I will fix on my end by querying the stop sequence.
On "normal" operation, it fails to find the "/n" resulting in last "-1".
But yeah, certain modes will trigger this behavior by default.
So trying another mode also solves the problem.

**Comment by VL4DST3R** [comment]
I don't think you should close this, maybe an official fix should/could be implemented since multiple people encountered this who won't know how or why it happens.

**Comment by jojorne** [comment]
Okay. So if you hit a certain combination, it will trigger `end_trim_to_sentence` and if the text starts with "/n", last will be 1, removing everything and leaving behind only the "\n". In my case, the text I have doesn't usually end with "\n", so the AI ​​always starts with "\n".
